tutorials/e2e-huggingface-summarization/e2e-dl-huggingface-summarization.ipynb

```diff
@@ -0,0 +1,1103 @@
+    "        dataset_config=dict(type=\"string\"),\n",
+    "        text_column=dict(type=\"string\"),\n",
+    "        summary_column=dict(type=\"string\"),\n",
+    "        max_samples=dict(type=\"integer\", default=-1),\n",
+    "    ),\n",
+    "    # The source folder of the component\n",
+    "    code=src_dir,\n",
+    "    # The environment the component job will be using\n",
+    "    environment=f\"{pipeline_job_env.name}:{pipeline_job_env.version}\",\n",
+    "    # The command that will be run in the component\n",
+    "    command=\"python baseline.py --dataset_name ${{inputs.dataset_name}} --dataset_config ${{inputs.dataset_config}} \"\n",
+    "    \"--text_column ${{inputs.text_column}} --summary_column ${{inputs.summary_column}} \"\n",
+    "    \"--max_samples ${{inputs.max_samples}}\",\n",
+    ")\n",
+    "\n",
+    "baseline_component = ml_client.create_or_update(baseline_component)\n",
+    "\n",
+    "print(\n",
+    "    f\"Component {baseline_component.name} with Version {baseline_component.version} is registered\"\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9edb5547",
+   "metadata": {},
+   "source": [
+    "### 1.2.1 Data Preparation Component\n",
+    "We can use a preprocessing step to save time on dataset download and tokenization. This is especially useful in scenarios involving Hyper Parameter Optimization."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d1417909",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# importing the CommandComponent Package\n",
+    "from azure.ai.ml.entities import CommandComponent\n",
+    "\n",
+    "src_dir = \"src/run_summarization/\"\n",
+    "\n",
+    "\n",
+    "data_prep_component = CommandComponent(\n",
+    "    # Name of the component\n",
+    "    name=\"data_prep_summarization\",\n",
+    "    # Component Version, no Version and the component will be automatically versioned\n",
+    "    # version=\"26\",\n",
+    "    # The dictionary of the inputs. Each item is a dictionary itself.\n",
+    "    inputs=dict(\n",
+    "        dataset_name=dict(type=\"string\"),\n",
+    "        dataset_config=dict(type=\"string\"),\n",
+    "        text_column=dict(type=\"string\"),\n",
+    "        summary_column=dict(type=\"string\"),\n",
+    "        max_samples=dict(type=\"integer\", default=-1),\n",
+    "        max_input_length=dict(type=\"integer\", optional=True, default=512),\n",
+    "        max_target_length=dict(type=\"integer\", optional=True, default=40),\n",
+    "        padding=dict(type=\"string\", optional=True, default=\"max_length\"),\n",
+    "        model_checkpoint=dict(type=\"string\"),\n",
+    "        source_prefix=dict(type=\"string\", optional=True, default=None),\n",
+    "    ),\n",
+    "    # The dictionary of the outputs. Each item is a dictionary itself.\n",
+    "    outputs=dict(\n",
+    "        encodings_datasets=dict(type=\"path\"),\n",
```

Comment:
nit: rename to just `encodings` so we avoid having "datasets" everywhere.

Reply:
done